FirebaseAuth/Sources/Swift/Utilities/AuthRecaptchaVerifier.swift

```diff
@@ -125,38 +125,83 @@
-              guard linked else {
-                continuation.resume(throwing: AuthErrorUtils.recaptchaSDKNotLinkedError())
-                return
-              }
-              guard actionCreated else {
-                continuation.resume(throwing: AuthErrorUtils.recaptchaActionCreationFailed())
-                return
-              }
-              if let error {
-                continuation.resume(throwing: error)
-                return
-              } else {
-                if token == "NO_RECAPTCHA" {
-                  AuthLog.logInfo(code: "I-AUT000031",
-                                  message: "reCAPTCHA token retrieval failed. NO_RECAPTCHA sent as the fake code.")
-                } else {
-                  AuthLog.logInfo(
-                    code: "I-AUT000030",
-                    message: "reCAPTCHA token retrieval succeeded."
-                  )
-                }
-                continuation.resume(returning: token)
-              }
-          }
-        }
+
+      let (token, error, linked, actionCreated) = await recaptchaToken(siteKey: siteKey, actionString: actionString, fakeToken: "NO_RECAPTCHA")
+
+      guard linked else {
+        throw AuthErrorUtils.recaptchaSDKNotLinkedError()
+      }
+      guard actionCreated else {
+        throw AuthErrorUtils.recaptchaActionCreationFailed()
+      }
+      if let error {
+        throw error
+      }
+      if token == "NO_RECAPTCHA" {
+        AuthLog.logInfo(code: "I-AUT000031",
+                        message: "reCAPTCHA token retrieval failed. NO_RECAPTCHA sent as the fake code.")
+      } else {
+        AuthLog.logInfo(
+          code: "I-AUT000030",
+          message: "reCAPTCHA token retrieval succeeded."
+        )
+      }
+      return token
       #endif // !(COCOAPODS || SWIFT_PACKAGE)
     }
 
+    private static var recaptchaClient: (any RCARecaptchaClientProtocol)?
+
+    private func recaptchaToken(
+      siteKey: String,
+      actionString: String,
+      fakeToken: String
+    ) async -> (token: String, error: Error?, linked: Bool, actionCreated: Bool) {
+      if recaptchaClient != nil {
+        return await retrieveToken(actionString: actionString, fakeToken: fakeToken)
+      }
+
+      if let recaptcha = NSClassFromString("RecaptchaEnterprise.RCARecaptcha") as? RCARecaptchaProtocol.Type {
+        do {
+          // TODO(ncooke3): This should be `fetchClient(withSiteKey:)`.
+          let client = try await recaptcha.getClient(withSiteKey: siteKey)
+          recaptchaClient = client
+          return await retrieveToken(actionString: actionString, fakeToken: fakeToken)
+        } catch {
+          return ("", error, true, true);
+        }
+      } else if let recaptcha = NSClassFromString("Recaptcha") {
+        // Fall back to attempting to connect with pre-18.7.0 RecaptchaEnterprise.
+          let client = try await recaptcha.getClient(withSiteKey: siteKey)
```

Comment:
This is mysteriously working locally when it technically shouldn't, because `recaptcha` is `AnyClass` here. This shouldn't even compile.

Reply:
Investigating an alternative #14217

Reply:
This led to some interesting learnings in [b570287](https://github.com/firebase/firebase-ios-sdk/pull/14201/commits/b570287473f8e7edf35c90912497afce67dfda2d), but I will remove the additional logic path it adds support for.